lightning/src/offers/offer.rs

```diff
@@ -468,6 +468,11 @@ impl Offer {
 		self.contents.is_expired()
 	}
 
+	/// Whether the offer has expired given the duration since the Unix epoch.
+	pub fn is_expired_no_std(&self, duration_since_epoch: Duration) -> bool {
```

Comment:
Could you add this to `Refund`? Also, would you mind updating `builds_offer_with_absolute_expiry` and `builds_refund_with_absolute_expiry` to call `is_expired_no_std` as well?